tools/wavetool.py

```diff
@@ -0,0 +1,157 @@
+import scipy.io.wavfile as wavefile
+import scipy.fftpack as fftpack
+
+eps = 0.0000000001 # used to avoid zero division error
+
+def generateSinusoid(**params):
+    func = np.sin
+    if params['type'] == 'cosine':
+        func = np.cos
+    time = np.arange(0, float(params['duration']), 1.0/float(params['sample_rate']), dtype=np.float32)
+    data = float(params['amp']) * func(2 * np.pi * float(params['freq']) * time + float(params['phase']))
+    return np.reshape(np.repeat(data, int(params['channel'])),[len(data), int(params['channel'])])
+
+def parse_cmdline():
+    parser = argparse.ArgumentParser(add_help=True, formatter_class=argparse.RawTextHelpFormatter,
+        description='A Tool to Generate and Manipulate Wave Files.')
+    parser.add_argument('-v', '--version', action='version', version='%(prog)s 1.0')
+    parser.add_argument('-g', '--generate', type=str, help='generate specified types of wave\n'
+    'parameter format: "wave_type; amplitude(0.0~1.0); frequency; phase; sample rate; channel; format; duration"\n'
+    'Note: please use float point precision\n'
+    'Supported wave type: sine, cosine\n'
+    'Support format: S8, S16, S32, F32\n')
+    parser.add_argument('-o', '--output', type=str, help='Path to store generated files', default='.')
+    parser.add_argument('-p', '--preset', type=str, help='Commonly used wave presets\n'
+    'Supported presets are: sine_1k_16bit_48k_10s')
+    parser.add_argument('-c', '--compare', type=str, help='compare recorded wave and reference wave\n'
+    'parameter format: "mode;recorded_wave_path; reference_wave_path"\n'
+    'Supported comparison modes are: HWEchoRef')
+    return vars(parser.parse_args())
+
+def parse_params_and_gen_wav(cmd_param):
+    supported_type = ['sine', 'cosine']
+    param_keys = ['type', 'amp', 'freq', 'phase', 'sample_rate', 'channel', 'format', 'duration']
+    split_cmd_params = cmd_param.split(';')
+    wave_params = [param.strip() for param in split_cmd_params]
+    param_dict = dict(zip(param_keys, wave_params))
+    if param_dict['type'] not in supported_type:
+        print("Unsupported wave type")
+        sys.exit(1)
+    if param_dict['type'] in ['sine', 'cosine']:
+        wave_data = generateSinusoid(**param_dict)
+        return wave_data, param_dict
+
+def do_wave_compare(cmd_args):
+    supported_mode = ['HWEchoRef']
+    cmd_line = cmd_args['compare'].strip().split(';')
+    comparison_mode = cmd_line[0].strip()
+    if comparison_mode not in supported_mode:
+        print("Unsupported comparison mode: %s" % comparison_mode)
+        print('Supported comparison modes are: ' + str(supported_mode))
+        sys.exit(1)
+    wav_path, ref_path = [path.strip() for path in cmd_line[1:]]
+    if not os.path.exists(wav_path):
+        print('Recorded wave path does not exist: %s' % wav_path)
+        sys.exit(1)
+    if not os.path.exists(ref_path):
+        print('Reference wave path does not exist: %s' % ref_path)
+        sys.exit(1)
```

Comment:
One big advantage of a high level language like Python over something like C is that you don't need checks like these which means one order of magnitude less code. You're losing this advantage. Just open the file and if not found then the built-in Python exception is perfectly readable (try it)